src/components/Registration/RegistrationContainer.tsx

```diff
@@ -93,6 +95,22 @@ export const RegistrationContainer = withRouter(
       setIsViewedDocumentsCheckboxChecked
     ] = useState(false);
 
+    const isAdministrationAlreadyRegistered =
```

Comment:
is this the only check we need between registration_status transitions ? if we expect to have other complex state transition logic mixed to ui state I'd suggest to take a look at https://xstate.js.org/docs/guides/start.html
and determine if it can be useful for us. 

(general concept is to switch from if-statements-logic (booleans) to a FSM)

Reply:
atm this is the only check on registration status,  but I will take a look at the link to understand these concepts